tests/bsim/bluetooth/audio/src/bap_broadcast_source_test.c

```diff
@@ -50,9 +55,135 @@ static struct bt_bap_lc3_preset preset_16_2_1 = BT_BAP_LC3_BROADCAST_PRESET_16_2
 static struct bt_bap_lc3_preset preset_16_1_1 = BT_BAP_LC3_BROADCAST_PRESET_16_1_1(
 	BT_AUDIO_LOCATION_FRONT_LEFT, BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED);
 
+static uint8_t bis_codec_data[] = {
+	BT_AUDIO_CODEC_DATA(BT_AUDIO_CODEC_CFG_CHAN_ALLOC,
+			    BT_BYTES_LIST_LE32(BT_AUDIO_LOCATION_FRONT_CENTER)),
+};
+
 static K_SEM_DEFINE(sem_started, 0U, ARRAY_SIZE(broadcast_source_streams));
 static K_SEM_DEFINE(sem_stopped, 0U, ARRAY_SIZE(broadcast_source_streams));
 
+static uint8_t count_bits(enum bt_audio_location chan_allocation)
+{
+	uint8_t cnt = 0U;
+
+	while (chan_allocation != 0) {
+		cnt += chan_allocation & 1U;
+		chan_allocation >>= 1;
+	}
+
+	return cnt;
+}
+
+static void validate_stream_codec_cfg(const struct bt_bap_stream *stream)
+{
+	const struct bt_audio_codec_cfg *codec_cfg = stream->codec_cfg;
+	const struct bt_audio_codec_cfg *exp_codec_cfg = &preset_16_1_1.codec_cfg;
```

Comment:
I'm not following this. 

The whole point of this change is that the stream->codec_cfg can be different from the `preset_16_1_1.codec_cfg` which is only applied to the subgroup. 

I strongly suggest that you attempt to write the test first and have it fail, and then apply your fix, and then verify that the test passes. 

I heavily assume that without your patch, this test still passes. 

This is again something that would have been much easier to verify as a unit test ;)

Reply:
As discussed, bis specific data is contained in bis_codec_data and contains only channel info (BT_AUDIO_LOCATION_FRONT_CENTER).  So channel info is verified to be bis specific data.

static uint8_t bis_codec_data[] = {
	BT_AUDIO_CODEC_DATA(BT_AUDIO_CODEC_CFG_CHAN_ALLOC,
			    BT_BYTES_LIST_LE32(BT_AUDIO_LOCATION_FRONT_CENTER)),
};

Rest of the codec data is the subgroup codec data which is preset_16_1_1.codec_cfg

Reply:
I think it's the naming that throws me off here. You are setting 
```
exp_codec_cfg = &preset_16_1_1.codec_cfg;
```
and the `preset_16_1_1` is what we use for the subgroups. 

We specifically do not expect the values of the subgroup, but rather the merged result of the subgroup and the BIS, and thus calling the subgroup's codec config for `exp_codec_cfg` does not sound right